### PR TITLE
Change/improve profiling

### PR DESCRIPTION
**Pull request by chobobobo**

### CODE REVIEW

**Comment by chobobobo** [comment]
this will resolve #1067

File: population.js [hunk 1]
```diff
@@ -249,28 +253,38 @@ mod.analyze = function(){
...
+    _.forEach(Memory.population, c => {
+        validateAssignment(c);
+        p.checkCPU('Validate: ' + c.creepName, PROFILING.ANALYZE_LIMIT / 2);
+    });
 };
 mod.execute = function(){
+    const p = startProfiling('Population.execute', {enabled: PROFILING.CREEPS});
     let triggerCompleted = name => Creep.spawningCompleted.trigger(Game.creeps[name]);
     this.spawned.forEach(triggerCompleted);
+    p.checkCPU('triggerCompleted', PROFILING.EXECUTE_LIMIT / 4);
```
Comment (karlthepagan) [inline population.js]: I don't want to require it but would precomputing all these divides / multiplies help? This might be my own superstition.

Comment (chobobobo) [inline population.js]: I think math is fairly well optimized, it just has to grab the constant and divide it right?  Fails the DRY test, so I will change it.

**Comment by chobobobo** [comment]
@Spedwards I have fixed Util and made use of it globally, as well as making some significant improvements.  I think this is ready to merge now.

File: util.js [hunk 2]
```diff
@@ -449,13 +459,9 @@ module.exports = {
                         Util.logSystem(name + ':' + localName, used);
                     }
                     if (type) {
-                        Util.set(global.profiler.types, type, {
-                            totalCPU: 0,
-                            count: 0,
-                            totalCount: 0,
-                        });
+                        if (_.isUndefined(global.profiler.types[type])) global.profiler.types[type] = {totalCPU: 0, count: 0, totalCount: 0};
```
Comment (Spedwards) [inline util.js]: Did ```Util.set(global.profiler, `types[${type}]`...``` not work?

Comment (chobobobo) [inline util.js]: It did not, I gave up in the end, the only issue is the . In the key, maybe I should change the flag types function to return type: subtype instead so we can use it

**Pull request merged**